### DISCUSSION TITLE
Filterlist update period below 1-day

### DISCUSSION THREAD

**Discussion by MasterKia** [uBlockOrigin/uBlock-issues#2544] [Q&A] [answered]
https://github.com/gorhill/uBlock/commit/db118483c91da468d22b943aba07bbcfc2e37427

What is the lowest possible amount and how to use it?
`! Expires: 1 hour`
`! Expires: 1/24 days`
?

**Comment by gothic-bum** [comment, marked as answer]
The minimum amount is 12 hours (0.5 day):  
https://github.com/gorhill/uBlock/blob/3c4c3767a2df806ec8e66183edca9038112870a1/src/js/storage.js#L974
  
Note the regex to extract update frequency information:  
https://github.com/gorhill/uBlock/blob/3c4c3767a2df806ec8e66183edca9038112870a1/src/js/storage.js#L966
  
As examples, the following are valid:  
  
    ! Expires: 12 hours
    !Expires: 12h
    # Expires: 12 h
    # Expires:12hours

**Reply by MasterKia** [reply]
```js
    const matches = head.match(/(?:^|\n)(?:!|# )[\t ]*Expires[\t ]*:[\t ]*(\d+)[\t ]*(h)?/i);
    if ( matches !== null ) {
        let updateAfter = parseInt(matches[1], 10);
        if ( isNaN(updateAfter) === false ) {
            if ( matches[2] !== undefined ) {
                updateAfter = Math.ceil(updateAfter / 12) / 2;
            }
            updateAfter = Math.max(updateAfter, 0.5);
```

I might be wrong but my understanding is that the regex assumes the `updateAfter` value is based on hours, not days (hence the division by 12: `updateAfter / 12`).
But we usually use `! Expires: 2 days` in lists and not `! Expires: 48 hours`.

**Reply by gothic-bum** [reply]
The regex checks for an `h` to indicate hours.

**Reply by MasterKia** [reply]
Then how does it deal with for example:
https://github.com/uBlockOrigin/uAssets/blob/master/filters/filters-2023.txt#L3

**Reply by gothic-bum** [reply]
It is in days, unless the `h` is extracted which then specifies hours:  
https://github.com/gorhill/uBlock/blob/3c4c3767a2df806ec8e66183edca9038112870a1/src/js/storage.js#L971,L972
